sql/core/src/main/scala/org/apache/spark/sql/execution/subquery.scala

```diff
@@ -171,6 +173,57 @@ case class InSubqueryExec(
   }
 }
 
+/**
+ * The physical node of non-correlated EXISTS subquery.
+ */
+case class ExistsExec(
+    plan: BaseSubqueryExec,
+    exprId: ExprId)
+  extends ExecSubqueryExpression {
+
+  @transient private var result: Boolean = _
```

Comment:
We should define the type as `Option[Boolean]`. The default value for boolean is false, not null.

Reply:
> The default value for boolean is false,

Forgot this point, can explain strange things I meet.